docs/data/data-grid/editing/FullFeaturedCrudGrid.tsx

```diff
@@ -101,6 +106,106 @@ function EditToolbar(props: EditToolbarProps) {
   );
 }
 
+interface ActionHandlers {
+  handleCancelClick: (id: GridRowId) => void;
+  handleDeleteClick: (id: GridRowId) => void;
+  handleEditClick: (id: GridRowId) => void;
+  handleSaveClick: (id: GridRowId) => void;
+}
+
+const ActionHandlersContext = React.createContext<ActionHandlers>({
+  handleCancelClick: () => {},
+  handleDeleteClick: () => {},
+  handleEditClick: () => {},
+  handleSaveClick: () => {},
+});
+
+function ActionsCell(props: GridRenderCellParams) {
+  const apiRef = useGridApiContext();
+  const rowModesModel = useGridSelector(apiRef, gridEditRowsStateSelector);
+  const isInEditMode = typeof rowModesModel[props.id] !== 'undefined';
+
+  const { handleSaveClick, handleCancelClick, handleEditClick, handleDeleteClick } =
+    React.useContext(ActionHandlersContext);
+
+  return (
+    <GridActionsCell
+      {...props}
+      actions={
```

Comment:
Could we pass `GridActionCellItem` as children instead? Could be a slightly nicer DX e.g. 
```tsx
<GridActionsCell {...props}>
  <GridActionsCellItem
    icon={<SaveIcon />}
    label="Save"
    sx={{
      color: 'primary.main',
    }}
    onClick={() => handleSaveClick(props.id)}
  />
  <GridActionsCellItem
    icon={<CancelIcon />}
    label="Cancel"
    className="textPrimary"
    onClick={() => handleCancelClick(props.id)}
    color="inherit"
  />
</GridActionsCell>
```